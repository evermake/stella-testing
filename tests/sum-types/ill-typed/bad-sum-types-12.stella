language core;
extend with #sum-types;

fn g(x : Nat + Bool) -> Nat {
  return match false {
      inr(bf) => if bf then 0 else succ(0) // Incorrect pattern for type Nat
    | inl(n) => succ(n) // Correct pattern for type Nat
  }
}

fn main(x : Nat) -> Nat {
  return g(inr(false))
}
