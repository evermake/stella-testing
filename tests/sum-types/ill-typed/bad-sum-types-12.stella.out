An error occurred during typechecking!
Type Error Tag: [ERROR_UNEXPECTED_PATTERN_FOR_TYPE]
unexpected pattern
  inr (bf)
when pattern matching is expected for type
  Bool